dataactbroker/handlers/fileHandler.py

```diff
@@ -758,13 +732,32 @@ def handleEmptyResponse(self, job, val_job):
 
     def get_xml_response_content(self, api_url):
         """ Retrieve XML Response from the provided API url """
-        _debug_logger.debug('Getting XML response')
+        logger.debug('Getting XML response')
         return requests.get(api_url, verify=False, timeout=120).text
 
-    def call_d_file_api(self, api_url):
+    def call_d_file_api(self, file_type_name, cgac_code, start_date, end_date, job):
         """ Call D file API, return True if results found, False otherwise """
-        # Check for numFound = 0
-        return "numFound='0'" not in self.get_xml_response_content(api_url)
+        file_type = FILE_TYPE_DICT_LETTER[FILE_TYPE_DICT[file_type_name]]
+        task_key = FileHandler.create_generation_task(job.job_id)
+
+        if not self.isLocal:
+            # Create file D API URL with dates and callback URL
+            api_url = FileHandler.get_d_file_url(task_key, file_type_name, cgac_code, start_date, end_date)
+
+            logger.debug('Calling Detached D file API => %s', api_url)
+            try:
+                # Check for numFound = 0
+                if not ("numFound='0'" not in self.get_xml_response_content(api_url)):
+                    sess = GlobalDB.db().session
+                    # If the call to the external API wasn't successful, mark the job as failed
+                    job.error_message = "%s data unavailable for the specified date range" % file_type
+                    job.job_status_id = JOB_STATUS_DICT['failed']
+                    sess.commit()
+            except Timeout as e:
+                exc = ResponseException(str(e), StatusCode.CLIENT_ERROR, Timeout)
+                return False, JsonResponse.error(e, exc.status, url="", start="", end="", file_type=file_type)
+        else:
+            self.complete_generation(task_key, file_type)
```

Comment:
This function returns different types depending on code path: `None` or `(False, error)`. Should it just return the error? If it's `None` we can assume there was no error.